spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -840,9 +840,9 @@ def interpreter_versions(self):
             import subprocess
             versions = {}
             pyexec = CONF.get('main_interpreter', 'executable')
-            py_cmd = '%s -c "import sys; print(sys.version)"' % pyexec
-            ipy_cmd = ('%s -c "import IPython.core.release as r; print(r.version)"'
-                       % pyexec)
+            py_cmd = 'env -i %s -c "import sys; print(sys.version)"' % pyexec

```

Comment:
I think this is the wrong way to do this and I really don't like it. Instead, env variables should be read using a subprocess or something and passed to the kernel in the kernelspec.

Reply:
I agree that env variables should be passed to the kernel through the kernelspec. These lines of code are doing exactly the opposite. The '-i' flag starts the environment entirely clean, no environment variables are inherited. I've placed these *only* for python invocations that are used to get versions or test if modules are installed. In these cases, you *don't* want to inherit Spyder's environment variables because it can cause incorrect results. In this particular case, only the IPython version needs to be retrieved and with the original command (sans 'env -i') results in the subprocess returning an error.

Reply:
I've moved these changes to #12235